.github/workflows/publish/action.yml

```diff
@@ -0,0 +1,115 @@
+        role-duration-seconds: 900
+
+    - name: Log in to Amazon public ECR
+      if: ${{ github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/') }}
+      shell: bash
+      run: aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws
+
+    - name: Import the PGP key
+      shell: bash
+      run: |
+        echo ${GPG_KEY_BASE64} | base64 -d > spacelift.gpg
+        gpg --import \
+          --passphrase=$GPG_PASSPHRASE \
+          --pinentry-mode=loopback \
+          spacelift.gpg
+        rm spacelift.gpg
+      env:
+        GPG_KEY_BASE64: ${{ inputs.gpg_base64_key }}
+        GPG_PASSPHRASE: ${{ inputs.gpg_passphrase }}
+    
+    - name: Sign Spacelift VCS Agent Binary
+      shell: bash
+      run: |
+        chmod 755 ./dist/spacelift-vcs-agent_linux_amd64_v1/spacelift-vcs-agent
+        ./scripts/sign.sh ./dist/spacelift-vcs-agent_linux_amd64_v1 spacelift-vcs-agent
+        ./scripts/verify.sh ./dist/spacelift-vcs-agent_linux_amd64_v1 spacelift-vcs-agent
+    
+        chmod 755 ./dist/spacelift-vcs-agent_linux_arm64/spacelift-vcs-agent
+        ./scripts/sign.sh ./dist/spacelift-vcs-agent_linux_arm64 spacelift-vcs-agent
+        ./scripts/verify.sh ./dist/spacelift-vcs-agent_linux_arm64 spacelift-vcs-agent
+      env:
+        GPG_KEY_ID: ${{ inputs.gpg_key_id }}
+        GPG_PASSPHRASE: ${{ inputs.gpg_passphrase }}
+    
+    - name: Put the files into the right place
+      shell: bash
+      run: |
+        mkdir -p build
+        cp dist/spacelift-vcs-agent_linux_amd64_v1/spacelift-vcs-agent build/spacelift-vcs-agent
+        cp dist/spacelift-vcs-agent_linux_amd64_v1/spacelift-vcs-agent_SHA256SUMS build/spacelift-vcs-agent_SHA256SUMS
+        cp dist/spacelift-vcs-agent_linux_amd64_v1/spacelift-vcs-agent_SHA256SUMS.sig build/spacelift-vcs-agent_SHA256SUMS.sig
+    
+        cp dist/spacelift-vcs-agent_linux_amd64_v1/spacelift-vcs-agent build/spacelift-vcs-agent-x86_64
+        cp dist/spacelift-vcs-agent_linux_amd64_v1/spacelift-vcs-agent_SHA256SUMS build/spacelift-vcs-agent-x86_64_SHA256SUMS
+        cp dist/spacelift-vcs-agent_linux_amd64_v1/spacelift-vcs-agent_SHA256SUMS.sig build/spacelift-vcs-agent-x86_64_SHA256SUMS.sig
+    
+        cp dist/spacelift-vcs-agent_linux_arm64/spacelift-vcs-agent build/spacelift-vcs-agent-aarch64
+        cp dist/spacelift-vcs-agent_linux_arm64/spacelift-vcs-agent_SHA256SUMS build/spacelift-vcs-agent-aarch64_SHA256SUMS
+        cp dist/spacelift-vcs-agent_linux_arm64/spacelift-vcs-agent_SHA256SUMS.sig build/spacelift-vcs-agent-aarch64_SHA256SUMS.sig
+    
+        # For visibility, here's the list of files we're uploading:
+        ls -l build
+    
+    - name: Upload the VCS Agent binaries to downloads.spacelift.[dev|io]
+      if: ${{ github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/') }}
+      shell: bash
+      run: aws s3 sync build/ s3://${{ inputs.aws_bucket }}
+    
```

Comment:
Probably I'm missing something here but where is container image build and push action? @peterdeme

Reply:
goreleaser @eminalemdar

Reply:
thank you! all clear now!